semantic-scope-graph/src/Control/Carrier/Sketch/Fresh.hs

```diff
@@ -0,0 +1,93 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-redundant-constraints #-}
+
+module Control.Carrier.Sketch.Fresh
+  ( SketchC (..)
+  , runSketch
+  , module Control.Effect.Sketch
+  ) where
+
+import           Control.Algebra
+import           Control.Carrier.Fresh.Strict
+import           Control.Carrier.State.Strict
+import           Control.Effect.Sketch
+import           Control.Monad.IO.Class
+import           Data.Bifunctor
+import           Data.Module
+import           Data.Name (Name)
+import qualified Data.Name
+import           Data.ScopeGraph (ScopeGraph)
+import qualified Data.ScopeGraph as ScopeGraph
+import           Data.Semilattice.Lower
+import           Source.Span
+import qualified System.Path as Path
+
+data Sketchbook address = Sketchbook
+  { sGraph        :: ScopeGraph address
+  , sCurrentScope :: address
+  } deriving (Eq, Show)
+
+instance Lower (Sketchbook Name) where
+  lowerBound =
+    let
+      initialGraph = ScopeGraph.insertScope n initialScope lowerBound
+      initialScope = ScopeGraph.Scope mempty mempty mempty
+      n = Data.Name.nameI 0
+    in
+      Sketchbook initialGraph n
+
+newtype SketchC address m a = SketchC (StateC (Sketchbook address) (FreshC m) a)
+  deriving (Applicative, Functor, Monad, MonadIO)
+
+instance forall address sig m . (address ~ Name, Effect sig, Algebra sig m) => Algebra (Sketch Name :+: sig) (SketchC Name m) where
```

Comment:
This quantifier is redundant, as is the equational constraint between `address` and `Name`.

```suggestion
instance (Effect sig, Algebra sig m) => Algebra (Sketch Name :+: sig) (SketchC Name m) where
```

Reply:
Yeah this is cruft. Fixing.